rust/toolchain.bzl

```diff
@@ -301,3 +301,170 @@ See @rules_rust//rust:repositories.bzl for examples of defining the @rust_cpuX r
 with the actual binaries and libraries.
 """,
 )
+
+def _cargo_tool_impl(ctx):
+    toolchain = ctx.toolchains[str(Label("//rust:toolchain"))]
+
+    # Executables must be produced within the rule marked executable. To
```

Comment:
Question - if rustfmt binary that we declare in the rust toolchain is executable, does it stay executable when we provide it in this rule through DefaultInfo? I honestly don't know.

One of the reasons why to make these rules executable is so they work as tools in genrules. Are there other reasons that you care about?

Reply:
I suppose they don't have to be executable but I think it's more intuitive since I (and I assume most people) would think of them as binaries and perhaps want to write rules consuming them via [ctx.executable](https://docs.bazel.build/versions/main/skylark/lib/ctx.html#executable). I could make these not executable if you think it's the smoother approach.

Reply:
I've updated the PR to have `toolchain_tool` and `toolchain_files`. Maybe it's not valuable to have an executable target? Thoughts and feelings?

Reply:
Ok, there's now no longer an executable which I think turns out to be way cleaner.